src/lib/mina_base/coinbase.ml

```diff
@@ -56,7 +56,11 @@ module Make_str (A : Wire_types.Concrete) = struct
       receiver t
       :: List.map ~f:Fee_transfer.receiver (Option.to_list t.fee_transfer)
     in
+    (* The order of this list will impact the order of new accounts in the ledger witness.
+       We use `List.rev` to have the same order than the tx application (in `apply_coinbase`)
+       where the "coinbase fee transfer" receiver is created before the "coinbase" receiver *)
     List.map account_ids ~f:(fun acct_id -> (acct_id, access_status))
+    |> List.rev
```

Comment:
Looks like this whole piece after `let access_status` definition can be simplified as something like:

```ocaml
let fee_transfer =
  let%map.Option transfer = t.fee_transfer in
  (Fee_transfer.receiver transfer, access_status)
in
Option.value_map ~default:ident ~f:List.cons fee_transfer [(receiver t, access_status)]
```

Current code is a bit hard to read
CC @mrmr1993